openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -12,6 +15,28 @@
 logger = logging.getLogger(__name__)
 
 
+class IngestionError(Exception):
+    """
+    Custom exception which includes information about the query_params that
+    were being used when the error was encountered.
+    """
+
+    def __init__(self, error, traceback, query_params):
+        self.error = error
+        self.traceback = traceback
+        self.query_params = json.dumps(query_params)
+
+    def __str__(self):
+        # Append query_param info to error message
+        return f"""{self.error}
+    query_params: {self.query_params}"""
+
+    def print_with_traceback(self):
```

Comment:
This doesn't quite print, which made it a bit confusing amidst the `logger.info` and `raise AirflowException` calls below. I can't really think of a better function name, maybe `repr_with_traceback`? 😅

Reply:
Thank you, this was bugging me but I couldn't come up with anything better! `repr_with_traceback` is a bit better, sold.